### PR TITLE
Introduced asDateTimeImmutable

### PR DESCRIPTION
**Pull request by veliu**

### CODE REVIEW

File: tests/ValueTest.php [hunk 1]
```diff
@@ -224,6 +225,34 @@ public function testFailAsNonEmptyStrings($raw): void
...
+        self::assertEquals($expected, $date->format(\DateTimeInterface::ATOM));
+    }
+
+    public static function dateTimeProvider(): iterable
+    {
+        yield [null, InvalidArgumentException::class, null];
+        yield [1234, InvalidArgumentException::class, null];
+        yield ['blub', TypeError::class, null];
+        yield ['20221111', null, '2022-11-11T00:00:00+00:00'];
+        yield ['2022-11-11', null, '2022-11-11T00:00:00+00:00'];
```
Comment (michaelpetri) [inline tests/ValueTest.php]: Can we add tests which include times and timezones?

File: src/Value.php [hunk 2]
```diff
@@ -138,4 +141,17 @@ public function asNonEmptyStrings(): array
 
         return array_values($this->value);
     }
+
+    public function asDateTimeImmutable(): DateTimeImmutable
```
Comment (michaelpetri) [inline src/Value.php]: IMO it makes sense to add an optional format here and use `createFromFormat`, what do you think?

Also please add new method to readme.

**Review by michaelpetri — CHANGES_REQUESTED** [review]

File: src/Value.php (hunk 2 above)
Comment (veliu) [reply src/Value.php]: Added :+1:

File: tests/ValueTest.php (hunk 1 above)
Comment (veliu) [reply tests/ValueTest.php]: Added

File: src/Value.php [hunk 3]
```diff
@@ -138,4 +142,23 @@ public function asNonEmptyStrings(): array
...
+    public function asDateTimeImmutable(?string $format = null): DateTimeImmutable
+    {
+        $value = $this->asNonEmptyString();
+
+        try {
+            $date = $format !== null
+                ? DateTimeImmutable::createFromFormat($format, $value)
+                : new DateTimeImmutable($value);
+        } catch (Exception $e) {
+            throw new TypeError(sprintf('"%s" is not a valid date string.', $value), 412, $e);
```
Comment (michaelpetri) [inline src/Value.php:154]: What's this 412 about?

File: src/Value.php [hunk 4]
```diff
@@ -138,4 +142,23 @@ public function asNonEmptyStrings(): array
...
+        }
+
+        if(!$date instanceof DateTimeImmutable) {
+            throw new TypeError(sprintf('Invalid values "%s" and format "%s"', $value, $format ?? 'no format'), 412);
```
Comment (michaelpetri) [inline src/Value.php]: Can we drop this no format and just print two different messages?

**Review by michaelpetri — APPROVED** [review]

File: src/Value.php (hunk 3 above)
Comment (veliu) [reply src/Value.php:154]: The code was required. So I thought a 412 does fit better than 0

Comment (michaelpetri) [reply src/Value.php:154]: Okay, let's keep it for now, i think this will get replaced by PSL anyway

File: src/Value.php (hunk 4 above)
Comment (veliu) [reply src/Value.php]: Splitted

**Pull request merged**